### PR TITLE
Update ignore trials stopping logic

### PR DESCRIPTION
**Pull request by micahwoodard**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
- add two training parameters auto_stop_ignore_win (default = 30) and auto_stop_ignore_ratio_threshold (default = 0.8) 
- after each trial, if session_time > 30 min and if  percentage of ignored on-autowater trials exceeds auto_stop_ignore_ratio_threshold, automatically stop the session
### What issues or discussions does this update address?
- resolves [#680](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/680)
### Describe the expected change in behavior from the perspective of the experimenter
- None
### Describe any manual update steps for task computers
- None
### Was this update tested in 446/447?
- [x] 446

### CODE REVIEW

**Comment by micahwoodard** [comment]
Probably buggy since I don't have a rig to test on but get's the general idea across.

File: src/foraging_gui/MyFunctions.py [hunk 1]
```diff
@@ -1166,26 +1166,26 @@ def _CheckStop(self):
         stop = False
         msg =''
         warning_label_text = ''
-        warning_label_color = 'color: gray;'        
 
         # Check for reasons to stop early 
-        if (np.shape(self.B_AnimalResponseHistory)[0]>=StopIgnore) and (np.all(self.B_AnimalResponseHistory[-StopIgnore:]==2)):
+        non_auto_reward = self.B_AnimalResponseHistory[np.where(~self.TP_AutoReward)]   # isolate non-auto-reward trials
+        win_sz = self.TP_auto_stop_ignore_win
+        if self.BS_CurrentRunningTime >= 30 and \
```
Comment (micahwoodard) [inline src/foraging_gui/MyFunctions.py:1173]: @hanhou is it okay that the session time check is hard coded to 30? Would there be any reason to make it a training parameter?

Comment (hanhou) [reply src/foraging_gui/MyFunctions.py:1173]: Making it as a parameter would be great

**Comment by micahwoodard** [comment]
Also just as a note to self, need to check formating on ephys ui since I can't see if full screen on my laptop

**Comment by micahwoodard** [comment]
Here is how the ephys gui will look like with the new parameters: 
image

**Comment by micahwoodard** [comment]
This is what the other gui looks like 
image

**Comment by XX-Yin** [comment]
@micahwoodard The logic looks good! Have you tested it in a session?

**Comment by hanhou** [comment]
Looks good to me too! Just a minor point: in the GUI, it might be better to put the "auto stop ignore window" field next to "auto stop ignore ratio threshold".

**Review by hanhou — APPROVED** [review]

**Comment by micahwoodard** [comment]
> @micahwoodard The logic looks good! Have you tested it in a session?

Yep just tested this morning

**Comment by micahwoodard** [comment]
> Looks good to me too! Just a minor point: in the GUI, it might be better to put the "auto stop ignore window" field next to "auto stop ignore ratio threshold".

And will do easy fix

**Review by XX-Yin — APPROVED** [review]

**Comment by micahwoodard** [comment]
@alexpiet @hanhou is there any concern that this will break post processing anywhere? Is B_StopIgnores being used anywhere?

**Comment by alexpiet** [comment]
I don't think so. 

FYI, this is the repo that does the downstream processing: https://github.com/AllenNeuralDynamics/aind-fip-nwb-base-capsule/tree/main

We should probably develop a better system such that changes to parameters in the GUI are automatically reflected in this repo, instead of having to manually make sure they are compatible. Probably something to adopt with the AIND Session/Rig/Task model

**Pull request merged**